packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,400 @@
+import { blockFromRpc } from '@ethereumjs/block/dist/from-rpc'
+import { RLP } from '@ethereumjs/rlp'
+import { Trie } from '@ethereumjs/trie'
+import {
+  Account,
+  arrToBufArr,
+  bigIntToHex,
+  bufferToHex,
+  intToHex,
+  isHexPrefixed,
+  toBuffer,
+} from '@ethereumjs/util'
+import { debug } from 'debug'
+import { hexToBytes } from 'ethereum-cryptography/utils'
+
+import { Cache } from './cache'
+
+import { BaseStateManager } from '.'
+
+import type { Proof, StateManager } from '.'
+import type { StorageDump } from './interface'
+import type { StorageProof } from './stateManager'
+import type { Common } from '@ethereumjs/common'
+import type { Address, PrefixedHexString } from '@ethereumjs/util'
+import type { JsonRpcProvider } from '@ethersproject/providers'
+
+const log = debug('statemanager')
+export interface EthersStateManagerOpts {
+  provider: JsonRpcProvider
+  blockTag?: bigint | string
```

Comment:
Should we change `string` to the literal `latest`| `earliest` | `pending` (or whichever ones we support) and then validate against those in the constructor (perhaps with an `unimplemented` error message for valid parameters that we don't yet support like we do for some RPC calls in the client)

Reply:
We certainly can and there shouldn't be any thing "unimplemented" here since we should expect the provider to support the RPC spec, right?

Reply:
Yeah you're right, think I wrote this before understanding the full picture of the ethersStateManager